Riot/Modules/Room/VoiceMessages/VoiceMessageToolbarView.swift

```diff
@@ -0,0 +1,332 @@
+    
+    @IBOutlet private var slideToCancelContainerView: UIView!
+    @IBOutlet private var slideToCancelLabel: UILabel!
+    @IBOutlet private var slideToCancelChevron: UIImageView!
+    @IBOutlet private var slideToCancelGradient: UIImageView!
+    
+    @IBOutlet private var lockContainerView: UIView!
+    @IBOutlet private var lockContainerBackgroundView: UIView!
+    @IBOutlet private var primaryLockButton: UIButton!
+    @IBOutlet private var secondaryLockButton: UIButton!
+    @IBOutlet private var lockChevron: UIView!
+    
+    @IBOutlet private var lockedModeContainerView: UIView!
+    @IBOutlet private var deleteButton: UIButton!
+    @IBOutlet private var playbackViewContainerView: UIView!
+    @IBOutlet private var sendButton: UIButton!
+    
+    private var playbackView: VoiceMessagePlaybackView!
+    
+    private var cancelLabelToRecordButtonDistance: CGFloat = 0.0
+    private var lockChevronToRecordButtonDistance: CGFloat = 0.0
+    private var lockChevronToLockButtonDistance: CGFloat = 0.0
+    private var panDirection: UISwipeGestureRecognizer.Direction?
+    
+    private var details: VoiceMessageToolbarViewDetails?
+    
+    private var currentTheme: Theme? {
+        didSet {
+            updateUIWithDetails(details, animated: true)
+        }
+    }
+    
+    weak var delegate: VoiceMessageToolbarViewDelegate?
+        
+    @objc static func instanceFromNib() -> VoiceMessageToolbarView {
+        let nib = UINib(nibName: "VoiceMessageToolbarView", bundle: nil)
+        guard let view = nib.instantiate(withOwner: nil, options: nil).first as? Self else {
+          fatalError("The nib \(nib) expected its root view to be of type \(self)")
+        }
+        return view
+    }
+    
+    override func awakeFromNib() {
+        super.awakeFromNib()
+        
+        slideToCancelGradient.image = Asset.Images.voiceMessageCancelGradient.image.withRenderingMode(.alwaysTemplate)
+        lockContainerBackgroundView.layer.cornerRadius = lockContainerBackgroundView.bounds.width / 2.0
+        
+        let longPressGesture = UILongPressGestureRecognizer(target: self, action: #selector(handleLongPress))
+        longPressGesture.delegate = self
+        longPressGesture.minimumPressDuration = 0.1
+        recordButtonsContainerView.addGestureRecognizer(longPressGesture)
+        
+        let panGesture = UIPanGestureRecognizer(target: self, action: #selector(handlePan))
+        longPressGesture.delegate = self
+        recordButtonsContainerView.addGestureRecognizer(panGesture)
+        
+        playbackView = VoiceMessagePlaybackView.instanceFromNib()
+        playbackViewContainerView.vc_addSubViewMatchingParent(playbackView)
+        
+        updateUIWithDetails(VoiceMessageToolbarViewDetails(), animated: false)
+    }
+    
+    func configureWithDetails(_ details: VoiceMessageToolbarViewDetails) {
+        elapsedTimeLabel.text = details.elapsedTime
+        
+        UIView.animate(withDuration: 0.25) {
```

Comment:
Do not hesitate to use constants for animation duration. We are using constants this in the project:
```
private enum Constants {
 static let detailsAnimationDuration: Timeinterval = 0.25
}
```